assets/js/curler.js

```diff
@@ -0,0 +1,76 @@
+
+  function RESTShorthandToCurl(el) {
+    var lines = el.textContent.split('\n');
+    var shorthand = /^(PUT|GET|POST|PATCH|DELETE)\s+(.+)/g.exec(lines[0]);
+    var out = [];
+    var curlOut = document.createElement('pre');
+    var curlCode = document.createElement('code');
+    var tabSelectors = {
+        divider : document.createElement('span'),
+        control : document.createElement('div'),
+        cURL : document.createElement('span'),
+        shorthand : document.createElement('span')
+    };
+    var tabActiveClass = 'active';
+    var tabNormalClass = 'curl-rest-control';
+    var tabPattern = [
+        ['none','block','remove','add'],
+        ['block','none','add','remove']
+    ];
+
+
+
+    if (shorthand) {
+        tabSelectors.control.classList.add('snippet-control-bar');
+        [tabSelectors.cURL,tabSelectors.shorthand].forEach(
+          function(controlEl) { controlEl.classList.add(tabNormalClass); }
+        );
+        curlOut.appendChild(curlCode);
+        curlOut.classList.add('highlight');
+        
+
+        tabSelectors.cURL.innerHTML = 'cURL';
+        tabSelectors.shorthand.innerHTML = 'REST';
+        tabSelectors.shorthand.classList.add(tabActiveClass);
+        tabSelectors.divider.innerHTML = ' &middot; ';
+        curlOut.style.display = 'none';
+
+
+
+        [tabSelectors.cURL, tabSelectors.shorthand].forEach(function(tabEl,index) {
+            tabEl.addEventListener('click', function () {
+                el.style.display = tabPattern[index][0];
+                curlOut.style.display = tabPattern[index][1];
+                tabSelectors.shorthand.classList[tabPattern[index][2]](tabActiveClass);
+                tabSelectors.cURL.classList[tabPattern[index][3]](tabActiveClass);
+            });
+        });
+       
+        out.push(
+            'curl \\',
+            '\t-H \'Content-Type: application/json\' \\',
```

Comment:
In general, I'm rabidly anti-tab. Sub two spaces here (and on subsequent lines)?